drgn_tools/partition.py

```diff
@@ -0,0 +1,99 @@
+# Copyright (c) 2023, Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/
+"""
+Helper to print partition information
+"""
+import argparse
+
+from drgn import Object
+from drgn import Program
+from drgn.helpers.common.format import escape_ascii_string
+from drgn.helpers.linux.block import for_each_partition
+from drgn.helpers.linux.block import part_devt
+from drgn.helpers.linux.block import part_name
+from drgn.helpers.linux.device import MAJOR
+from drgn.helpers.linux.device import MINOR
+
+from drgn_tools.corelens import CorelensModule
+from drgn_tools.table import print_table
+
+
+def get_partinfo_from_blkdev_struct(part: Object) -> list:
+    """
+    Collect partition information from `struct block_device`
+    Returns a list with partition information for the given partition.
+    """
+    devt = part.bd_dev.value_()
+    name = escape_ascii_string(part_name(part), escape_backslash=True)
+    start_sect = int(part.bd_start_sect)
+    nr_sects = int(part.bd_inode.i_size.value_() / 512)
+    if part.bd_read_only:
+        ro = "Yes"
+    else:
+        ro = "No"
+    return [
+        MAJOR(devt),
+        MINOR(devt),
+        name,
+        start_sect,
+        nr_sects,
+        ro,
+        hex(part.value_()),
+    ]
+
+
+def get_partinfo_from_hd_struct(part: Object) -> list:
+    """
+    Collects partition information from `struct hd_struct`
+    Returns a list with partition information for the given partition.
+    """
+    devt = part_devt(part)
+    name = part_name(part).decode()
```

Comment:
Wonderding why using "escape_ascii_string(part_name(part), escape_backslash=True)" in "get_partinfo_from_blkdev_struct" but using part_name(part).decode() here?

Reply:
No specific reason junxiao, i'm making both of use decode to make things consistent